### ISSUE TITLE
hasMoreResults of changefeed iterator not working as expected

### ISSUE THREAD

**Issue by totodile94** [Azure/azure-cosmos-js#244]
**Describe the bug**
hasMoreResults of change-feed iterator always returns true . This happens even after all changes are consume via change feed . 

**Package-name**: **azure/cosmos**
**Package-version**: 2.1.2

**Comment by christopheranderson** [comment]
Could you share your ChangeFeedOptions you used? There are some cases where we might return empty pages, but not indicate that there are no more results.

Do note that even if you do see hasMoreResults == false, you could executeNext and get more results if more things are updated since the last request.

**Comment by totodile94** [comment]
@christopheranderson   Regarding changeFeedOptions, I set startFromBeginning to true. 

My code was trying to do something similar to 

```
do {
      do stuff
 } while (changeFeedIterator.hasMoreResults)
```

Even after consuming all the results , the loop never breaks.

**Comment by christopheranderson** [comment]
Ok. I'll try to repro from there then.

**Comment by christopheranderson** [comment]
Here's a full repro: https://gist.github.com/christopheranderson/4b444b1090c1147dfe9ce676df4001b4

Main snippet:
```js
//@ts-check
const cosmos = require("@azure/cosmos");
const {items} = require("./sampleItems");

process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;

const {CosmosClient} = cosmos;

const endpoint = process.env.COSMOS_ENDPOINT;
const key = process.env.COSMOS_KEY;
const client = new CosmosClient({endpoint, key});

async function main() {
    const container = await init();
    
    const cfquery = container.items.readChangeFeed({ startFromBeginning: true});

    const maxpages = 20;
    let currentpage = 0;
    do {
        const page = await cfquery.executeNext();
        console.log(`Read ${page.count} items`);
        if(++currentpage >= maxpages) {
            throw new Error("Infinite loop detected");
        }
    } while(cfquery.hasMoreResults)
    return "done";
}

async function init() {
    const {database: db} = await client.databases.createIfNotExists({
        id: "change-feed-test"
    });
    const {container} = await db.containers.createIfNotExists({
        id: "chrange-feed-test"
    });
    for(const item of items) {
        await container.items.upsert(item);
    }
    return container;
}

main().then(console.log).catch(console.error);
```

**Comment by christopheranderson** [comment]
Status code is getting dropped here which means we never see it succeeding. Prepping a bug fix now.

https://github.com/Azure/azure-cosmos-js/blob/master/src/ClientContext.ts#L265-L276

**Comment by christopheranderson** [comment]
This will be fixed in the next release. I'll probably do a release today or Monday.

**Comment by southpolesteve** [comment]
This has been released in 2.1.4

**Comment by totodile94** [comment]
@christopheranderson  @southpolesteve Thanks for the quick fix
